pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -312,7 +312,17 @@ def save_checkpoint(self, checkpoint: Dict[str, Any], filepath: _PATH) -> None:
             checkpoint: dict containing model and trainer state
             filepath: write-target file's path
         """
-        return self.checkpoint_io.save_checkpoint(checkpoint, filepath)
```

Comment:
save_checkpoint() is same as training_type_plugin, should we remove it?

Reply:
rank check is different (default global_zero v.s. all ranks here)

Reply:
@jjenniferdai the logic difference is in XLACheckpointIO.save_checkpoint() right? the strategy overrides here is same as training_type_plugin so it's not needed?

Reply:
yep IO differences in `checkpointIO` but there's also different rank logic controlled by TTPs hence the TTP override:

training_type_plugin: `if global_zero: checkpointIO.save()`
here: `checkpointIO.save()` (all ranks)